### DISCUSSION TITLE
about Robidoux Sharp Cylindrical Filter

### DISCUSSION THREAD

**Discussion by zhouwangpei123** [ImageMagick/ImageMagick#5584] [Help] [answered]
where can i find the implementation of Robidoux Sharp Cylindrical Filter? I can only find the code of Robidoux Sharp orthogonal filter in the resize.c file.

**Comment by snibgo** [comment]
I'm not sure what you are asking.

"-resize" operates with two orthogonal filters. Code for this is in MagickCore\resize.c.

"-distort resize" uses a single cylindrical filter. Code for this is in MagickCore\distort.c,

**Reply by zhouwangpei123** [reply]
thanks for the reply. "-distort resize uses a single cylindrical filter" as you mentionend， is this cylindrical filter the well-known EWA Robidoux or EWA RobidouxSharp?

**Comment by snibgo** [comment, marked as answer]
"Robidoux" and "RobidouxSharp" are both filters. Like other filters, they can be used as orthogonal filters ("-resize") or cylindrical (EWA) filters ("-distort resize").

For example:
```
magick toes.png -filter Robidoux -resize 200% x1.png

magick toes.png -filter RobidouxSharp -resize 200% x2.png

magick toes.png -filter Robidoux -distort resize 200% x3.png

magick toes.png -filter RobidouxSharp -distort resize 200% x4.png
```
The four results are slightly different.

**Reply by zhouwangpei123** [reply]
that helps, thanks
